examples/bri-3/src/bri/identity/bpiSubjectAccounts/agents/bpiSubjectAccounts.agent.ts

```diff
@@ -17,20 +17,44 @@ export class BpiSubjectAccountAgent {
 
   public async getBpiSubjectAccountsAndThrowIfNotExist(
     subjectAccountIds: string[],
-  ) {
+  ): Promise<BpiSubjectAccount[]> {
     const subjectAccounts: BpiSubjectAccount[] = [];
 
     for (const subjectAccountId of subjectAccountIds) {
       const subjectAccount =
         await this.subjectAccountStorageAgent.getBpiSubjectAccountById(
           subjectAccountId,
         );
+
+      if (subjectAccount == null) {
+        throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+      }
+
       subjectAccounts.push(subjectAccount);
     }
 
     return subjectAccounts;
   }
 
+  public async getBpiSubjectAccounts(
+    subjectAccountIds: string[],
+  ): Promise<BpiSubjectAccount[]> {
+    const subjectAccounts: BpiSubjectAccount[] = [];
+
+    for (const subjectAccountId of subjectAccountIds) {
```

Comment:
nit but shouldn't this be one query instead of one query per id?

Reply:
https://github.com/eea-oasis/baseline/issues/697